### DISCUSSION TITLE
auto-budget with correction for overspending not adding amount correctly

### DISCUSSION THREAD

**Discussion by kuroshiropanda** [firefly-iii/firefly-iii#8641] [Firefly III]
### Support guidelines

- [X] I've read the [support guidelines](https://github.com/firefly-iii/firefly-iii/blob/main/.github/support.md)

### I've found a bug and checked that ...

- [X] ... [the documentation](https://docs.firefly-iii.org/) does not mention anything about my problem
- [X] ... there are no open or closed issues that are related to my problem
- [X] ... it's [definitely a Firefly III issue, not me](https://github.com/firefly-iii/firefly-iii/blob/main/.github/its_you_not_me.md)

### Description

![image](https://github.com/firefly-iii/firefly-iii/assets/17986144/aa66a712-502f-49ca-ba49-abe5b7f6a789)
I have a budget wherein it should add 500 monthly but when the 1st day of the month comes the auto-budget amount didn't add correctly
![image](https://github.com/firefly-iii/firefly-iii/assets/17986144/4a4a54ce-6e9b-45f9-a132-83437ef63f88)
budget for march is currently 1086.14 instead of 1205.14 (1029.63 (feb budget) - 324.49 (feb expenses) = 705.14 (budget left for feb) + 500 (march) = 1205.14)
transactions from february wasn't added late so it should have computed it correctly when the cron ran

### Debug information

Debug information generated at 2024-03-06 14:09:44 for Firefly III version **v6.1.9**.

<table><thead><tr><th colspan="2">System information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Firefly III</td><td>6.1.9 / v2.0.12 / 22 (exp. 22)</td></tr><tr><td>PHP version</td><td>8.3.2 / apache2handler / Linux</td></tr><tr><td>BCscale</td><td>12</td></tr><tr><td>Error reporting</td><td>Display: Off, reporting: ALL errors</td></tr><tr><td>Max upload</td><td>67108864 (64 MB)</td></tr><tr><td>Database drivers</td><td><strong>*mysql*</strong>, pgsql, sqlite, </td></tr><tr><td>Docker build</td><td><span>#</span>957, base <span>#</span>79</td></tr></tbody></table><table><thead><tr><th colspan="2">Firefly III information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>Timezone</td><td>Asia/Manila + Asia/Manila</td></tr><tr><td>App environment</td><td>local, debug: false</td></tr><tr><td>Layout</td><td>v1</td></tr><tr><td>Logging</td><td>notice, stack / (empty)</td></tr><tr><td>Cache driver</td><td>redis</td></tr><tr><td>Default language and locale</td><td>en_US + equal</td></tr><tr><td>Trusted proxies</td><td>**</td></tr><tr><td>Login provider & user guard</td><td>eloquent / web</td></tr><tr><td>Login headers</td><td>N/A + N/A</td></tr><tr><td>Stateful domains</td><td></td></tr><tr><td>Last cron job</td><td>2024-03-06 00:00:00 (14 hours ago)</td></tr><tr><td>Mailer</td><td>smtp</td></tr></tbody></table><table><thead><tr><th colspan="2">User-specific information</th></tr><tr><th>Item</th><th>Value</th></tr></thead><tbody><tr><td>User</td><td><span>#</span>1 of 1</td></tr><tr><td>User flags</td><td><span title="Has liabilities">:credit_card:</span> <span title="Has piggy banks">:pig:</span> <span title="Has rules">:wrench:</span> <span title="Has recurring transactions">:clock130:</span> <span title="Has object groups">:bookmark_tabs:</span> <span title="Has subscriptions">:email:</span></td></tr><tr><td>Session start</td><td>2024-03-01 00:00:00</td></tr><tr><td>Session end</td><td>2024-03-31 23:59:59</td></tr><tr><td>View range</td><td>1M</td></tr><tr><td>User language</td><td>en_US</td></tr><tr><td>User locale</td><td>en_US</td></tr><tr><td>Locale(s) supported</td><td>en_US.utf8: :white_check_mark:<br>en_US.UTF-8: :white_check_mark:<br></td></tr><tr><td>User agent</td><td>Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36 Edg/122.0.0.0</td></tr></tbody></table>

**Comment by JC5** [comment]
When I run this scenario manually, I end up at the correct amount.

```
local.DEBUG: Will now manage rollover for auto budget #4
local.DEBUG: Current period is 2024-03-01-2024-03-31, so previous period is 2024-02-01-2024-02-29
local.DEBUG: Going to find a budget limit for budget #7 ("Test") between 2024-02-01 and 2024-02-29
local.DEBUG: Budget limit exists for previous period.
local.DEBUG: GroupCollector: postFilterCollection has 0 filter(s) and 1 transaction(s).
local.DEBUG: Spent in previous budget period (2024-02-01-2024-02-29) is -324.490000000000
local.DEBUG: Total amount available for current budget period is 1205.140000000000
local.INFO: There is no overspending, no need to adjust. Budget limit amount will be 1205.140000000000.
local.DEBUG: No budget limit exist. Must create one for auto-budget #4
local.DEBUG: Amount is overruled and will be set to 1205.140000000000
```

What happens when you delete the amount for March, and run the cron again?

Delete the amount for March on the budgets page:

<img width="730" alt="Screenshot 2024-03-06 at 07 28 07" src="https://github.com/firefly-iii/firefly-iii/assets/5889984/f484c743-2dc0-47ab-85e7-a89407b0652b">

Run the cron again:

```
docker exec -it bash firefly php artisan firefly:cron --force --date=2024-03-01
```

**Comment by kuroshiropanda** [comment]
that worked correctly. hmm.. any insights on what could have caused the wrong amount?

cron runs via docker included in the docker compose

**Reply by JC5** [reply]
Not a clue! I'd have to see the original logs to see what happened. My best guess is a backdated transaction, that's usually the cause.
